core/src/avm2/e4x.rs

```diff
@@ -524,7 +529,19 @@ impl<'gc> E4XNode<'gc> {
     }
 
     pub fn descendants(&self, name: &Multiname<'gc>, out: &mut Vec<E4XOrXml<'gc>>) {
-        if let E4XNodeKind::Element { children, .. } = &self.0.read().kind {
+        if let E4XNodeKind::Element {
+            children,
+            attributes,
+            ..
+        } = &self.0.read().kind
+        {
+            if name.is_attribute() {
+                for attribute in attributes {
+                    if attribute.matches_name(name) {
+                        out.push(E4XOrXml::E4X(*attribute));
+                    }
+                }
+            }
             for child in children {
                 if child.matches_name(name) {
```

Comment:
if we did `if !name.is_attribute()` here, then `matches_name` itself doesn't need to do the is-attribute check anymore, I think